hal/src/rtl872x/ble_hal.cpp

```diff
@@ -2381,6 +2381,7 @@ void BleGap::handleConnectionStateChanged(uint8_t connHandle, T_GAP_CONN_STATE n
                 LOG_DEBUG(TRACE, "Connected as Central");
             } else {
                 connection.info.role = BLE_ROLE_PERIPHERAL;
+                connecting_ = false;
```

Comment:
Should the check at https://github.com/particle-iot/device-os/pull/2839/files#diff-f291a584aa5be86f324f9fa672da24a8055116bbe0194745178ab5eb74ce25f7R2403 just handle both cases once we generate appropriate events etc?

Reply:
When the device is acting as BLE peripheral, the device won't initiate the ATT MTU updating process, unless the BLE central device initiates. Under some circumstances, the BLE central device may omit this process. But for Gen4 platform that acts as BLE central role, from what we observed, it always initiates the process.